common/app/model/dotcomrendering/pageElements/CartoonExtraction.scala

```diff
@@ -0,0 +1,47 @@
+package model.dotcomrendering.pageElements
+
+import com.gu.contentapi.client.model.v1.{CartoonElementFields, CartoonImage}
+import model.ImageAsset
+
+object CartoonExtraction {
+  def extractCartoon(cartoonElementFields: CartoonElementFields): CartoonBlockElement = {
+    CartoonBlockElement(
+      variants = getCartoonVariants(cartoonElementFields),
+      role = Role(cartoonElementFields.role, Inline),
+      credit = cartoonElementFields.credit,
+      caption = cartoonElementFields.caption,
+      alt = cartoonElementFields.alt,
+      displayCredit = cartoonElementFields.displayCredit,
+    )
```

Comment:
Small thing, but given that we now have some conditions that `CartoonBlockElement` has to meet to be valid, I wonder if it would make sense to have `extractCartoon` return an `Option[CartoonBlockElement]` and handle the validation itself? That way you can never create a `CartoonBlockElement` that's invalid.

Then in `PageElement.scala` you can do:

```scala
element.cartoonTypeData.flatMap(extractCartoon)
```

in place of `map(...).filter(...)`, and likewise in the test file.

```suggestion
  def extractCartoon(cartoonElementFields: CartoonElementFields): Option[CartoonBlockElement] = {
    val variants = getCartoonVariants(cartoonElementFields)
    
    Option.when(cartoonIsValid(variants)) {
      CartoonBlockElement(
        variants = variants,
        role = Role(cartoonElementFields.role, Inline),
        credit = cartoonElementFields.credit,
        caption = cartoonElementFields.caption,
        alt = cartoonElementFields.alt,
        displayCredit = cartoonElementFields.displayCredit,
      )
    }
  }
```

Reply:
Thank you for the suggestion! This has been applied here: https://github.com/guardian/frontend/pull/26439/commits/a3e2e3ac40984a4d9ae64a8eed85f0f92f65765e